qa/os/src/test/java/org/elasticsearch/packaging/util/Packages.java

```diff
@@ -240,6 +241,45 @@ private static void verifyDefaultInstallation(Installation es, Distribution dist
 
         Stream.of("users", "users_roles", "roles.yml", "role_mapping.yml", "log4j2.properties")
             .forEach(configFile -> assertThat(es.config(configFile), file(File, "root", "elasticsearch", p660)));
+        verifySecurityAutoConfigured(es, distribution);
+    }
+
+    private static void verifySecurityAutoConfigured(Installation es, Distribution distribution) throws IOException {
+        assertThat(es.config("auto_generated_certs"), file(Directory, "root", "elasticsearch", p755));
+        Stream.of("http_keystore.p12", "http_truststore.p12", "transport_keystore_all_nodes.p12", "transport_truststore_all_nodes.p12")
+            .forEach(
+                keystore -> assertThat(es.config("auto_generated_certs").resolve(keystore), file(File, "root", "elasticsearch", p660))
+            );
+        List<String> configLines = Files.readAllLines(es.config("elasticsearch.yml"));
+        assertThat(configLines, contains("xpack.security.enabled: true"));
+        assertThat(configLines, contains("xpack.security.enrollment.enabled: true"));
+        assertThat(configLines, contains("xpack.security.authc.realms.file.auto_generated_1625753263.order: 0"));
```

Comment:
Is the realm name stable?

Reply:
No, I'll need to adjust the tests once we finalize #74868 , I just wanted to have a skeleton for the tests here too. Happy to get ideas about other Packaging tests that would make sense to add!